frontend/src/common/hooks/hooks.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from "@tanstack/react-query";
+import { httpGETRequest } from "../apiManager/httpRequestHandler";
+import { VEHICLES_URL } from "../apiManager/endpoints/endpoints";
+
+/**
+ * Fetch payment information by transaction id.
+ * @returns PaymentTransaction data as response, or null if fetch failed
+ */
+export const getFeatureFlags = async () : 
+  Promise<Record<string, string>> => {
+    const url = `${VEHICLES_URL}/feature-flags`;
+    return httpGETRequest(url).then((response) => response.data);
+};
+
+/**
+ * A custom react query hook that fetches applications in progress.
+ * @returns List of applications in progress
+ */
+export const useFeatureFlagsQuery = () => {
+    return useQuery({
+        queryKey: ["featureFlags"],
```

Comment:
I think it is worthwhile to set `stateTime` to infinity and `refetchInterval` to false so that this API call basically happens only once and the rest of the app, wherever used, only pulls data from the query Client cache.

Reply:
thanks - I'll make those updates. Appreciate it.